triedb/hashdb/database.go

```diff
@@ -99,6 +99,7 @@ type cache interface {
 type Config struct {
 	CleanCacheSize int    // Maximum memory allowance (in bytes) for caching clean nodes
 	StatsPrefix    string // Prefix for cache stats (disabled if empty)
+	ReferenceRoot  bool   // Whether to reference the root node on update
```

Comment:
How about `AtomicReferenceRootOnUpdate` to explain exactly what it does? As a newcomer this was one of the more frustrating modifications because it seemed bigger as I had to drill down a few layers before realising that it was something quite simple.

Reply:
I'm going with ReferenceRootOnUpdateAtomically to avoid confusion with the AtomicTrie